actpool/actpool.go

```diff
@@ -150,6 +150,17 @@ func (ap *actPool) Add(act action.SealedEnvelope) error {
 	if !ap.enableExperimentalActions && action.IsExperimentalAction(act.Action()) {
 		return errors.New("Experimental action is not enabled")
 	}
+	// Reject action if action source address is blacklisted
+	pubKeyHash := act.SrcPubkey().Hash()
+	srcAddr, err := address.FromBytes(pubKeyHash)
+	if err != nil {
+		return errors.Wrap(err, "failed to get address from bytes")
+	}
+	for _, addr := range ap.cfg.BlackList {
```

Comment:
when new actPool, load ap.cfg.BlackList as a map. This way the lookup time is always constant. now is O(n)